src/pkg/archive/archive.go

```diff
@@ -135,6 +136,19 @@ type CompressOpts struct{}
 
 // Compress takes any number of source files and archives them into a compressed archive at dest path.
 func Compress(ctx context.Context, sources []string, dest string, _ CompressOpts) (err error) {
+	if len(sources) == 0 {
+		return fmt.Errorf("sources cannot be empty")
+	}
+	if dest == "" {
+		return fmt.Errorf("dest cannot be empty")
+	}
+
+	// Ensure dest parent directories exist
+	err = os.MkdirAll(filepath.Dir(dest), helpers.ReadExecuteAllWriteUser)
```

Comment:
nit: we have a package const `dirPerm` that we have been using to help drive package isolation given this is not only an internal library. I believe in doing so we adhere closely to [dependencies in go](https://go.dev/talks/2012/splash.article#TOC_7.) for imports and help prevent potential consumer issues for cyclic imports.